format-parquet/src/main/java/io/cdap/plugin/format/parquet/input/ParquetInputFormatProvider.java

```diff
@@ -80,6 +80,15 @@ public Schema detectSchema(FormatContext context, InputFiles inputFiles) throws
       }
       try (ParquetFileReader parquetFileReader = ParquetFileReader.open(new HadoopInputFile(inputFile), readOptions)) {
         MessageType parquetSchema = parquetFileReader.getFooter().getFileMetaData().getSchema();
+        if (conf.getPathField() != null) {
+          List<Type> fields = new ArrayList<>();
+          for (Type t : parquetSchema.getFields()) {
+            fields.add(t);
+          }
+          fields.add(Types.primitive(PrimitiveType.PrimitiveTypeName.BINARY, Type.Repetition.OPTIONAL).as
```

Comment:
instead of adding a parquet field, convert the parquet schema into a CDAP schema first, then add the field as a CDAP field.